Be careful with the LDFLAGS ordering, so --as-needed works correctly.

diff --git i/Makefile w/Makefile
index dd03cab..389bf83 100644
--- i/Makefile
+++ w/Makefile
@@ -7,8 +7,8 @@ RM        = rm -f
 CFLAGS    += -DSHAPE # comment out this line to disable shape support
 LDFLAGS   += -lXext # if shape support is disabled, this normally is no longer necessary and schould probally be commented out too
 #CFLAGS    += -DDEBUG -ggdb3 # -DDEBUG_EVENTS -DSYNC # for debugging
-CFLAGS    += -I$(XROOT)/include
-LDFLAGS   += -lX11 -L$(XROOT)/lib
+CFLAGS    +=
+LDFLAGS   += -lX11
 CFLAGS    += -DVERSION=\"0.0.90\"
 
 SRCFILES  := actions.c buttons.c client.c config.c drag.c events.c evn.c ewmh.c info.c input.c main.c misc.c vdesk.c wlist.c x11.c
@@ -17,7 +17,7 @@ OBJFILES  := $(SRCFILES:.c=.o)
 all: matwm2
 
 matwm2: $(OBJFILES)	
-	$(CC) $(LDFLAGS) $(OBJFILES) -o $@
+	$(CC) $(OBJFILES) -o $@ $(LDFLAGS)
 
 clean:
 	$(RM) matwm2 *.o
